vllm/attention/layer.py

```diff
@@ -66,11 +66,21 @@ def __init__(
             block_size = cache_config.block_size
             is_attention_free = cache_config.is_attention_free
             calculate_kv_scales = cache_config.calculate_kv_scales
+            use_v1 = cache_config.use_v1
         else:
             kv_cache_dtype = "auto"
             block_size = 16
             is_attention_free = False
             calculate_kv_scales = False
+            if not envs.is_set("VLLM_USE_V1"):
+                raise NotImplementedError(
+                    "VLLM_USE_V1 is not set and your model does"
+                    "not pass a CacheConfig to the Attention layer. "
+                    "As a result, we cannot detect whether to use V0 "
+                    "or V1 attention. Please set VLLM_USE_V1=0 or "
+                    "VLLM_USE_V1=1 and file an issue on Github so "
+                    "we can address it.")
```

Comment:
Are there known cases where this will happen, or is this paranoia?

Reply:
paranoia